doc/unicode.md

```diff
@@ -0,0 +1,121 @@
+
+- Latin-1 (or other non UTF-8) sources compiled with -gnatW8 will likely cause "illegal wide character" errors, as Latin-1 literals will be invalid UTF-8 sequences.
+- UTF-8 sources compiled without -gnatW8 will result in mangled output as the UTF-8 sequences are misinterpreted as Latin-1, where most bytes are a valid character code.
+- Latin-1 sources saved to UTF-8 and compiled with -gnatW8 may result in "literal out of range of type Standard.Character" errors when characters outside of Latin-1 are assigned to a regular `Character` instead of `Wide_Wide_Character`.
+    - Example: `X : String := "€";`
+        - Will work without `-gnatW8`, resulting in a three-byte string.
+        - Will err with `-gnatW8`, as '€' is out of `Character` range.
+        - In this case, the code has to be adapted to use `Wide_Wide_String`/`Wide_Wide_Character`, or the string manually encoded as UTF-8 (which will **not** be apt for use with `Ada.Text_IO`):
+        `subtype UTF8_String is String;`
+        `X : UTF8_String := Ada.Strings.UTF_Encoding.Wide_Wide_Strings.Encode ("€");`
+
+
+See a few more examples:
+
+```
+O_Acute : constant Character := 'ó';
+--  With Latin-1 source and -gnatW8, it will fail as 'ó' won't be a proper UTF-8 sequence in the input file.
+--  With UTF-8 source and -gnatW8, it will work **but** it will be stored in Latin-1 encoding in memory.
+--  It can be converted to a Wide_Wide_String (that will be in UTF-32)
+--  with `Ada.Characters.Conversions.To_Wide_Wide_String`. In this case,
+--  using `Ada.Strings.UTF_Encoding.Wide_Wide_Strings.Decode` would be wrong,
+--  as the String is Latin-1 and not UTF-8 despite the source file encoding being UTF-8.
+
+Bye : constant String := "Adiós";
+--  Without -gnatW8 but UTF-8 sources without a BOM marker, this will end being an UTF-8 byte sequence,
+--  as it is interpreted literally as Latin-1 characters.
+--  With -gnatW8, the UTF-8 will be properly loaded and converted to Latin-1 for the in-memory representation.
+
+Bye : constant String (1 .. 5) := "Adiós";
+--  This is proper for a Latin-1-encoded file and compiling without -gnatW8.
+--  With -gnatW8 and an UTF-8 file, it will work properly and also end as a 5-byte Latin-1 string.
+--  With -gnatW8 but with Latin-1-encoded file, it will fail as the sequence won't be proper UTF-8.
+--  Without -gnatW8 but with an UTF-8 file it will fail as the sequence will be 6 bytes long.
+
+Euro : constant String := "€";
+--  Without -gnatW8 and a UTF-8-encoded file, this results in a 3-byte UTF-8 string.
+--  With -gnatW8 and a UTF-8-encoded file, this results in an error as '€' is outside of Character.
+```
+
+## Unicode in identifiers
+
+By default, GNAT only allows Latin-1 identifiers. To enable full Unicode identifiers you can use `-gnatiw`. This is generally not recommended, as that may greatly difficult others to work with your code. Alire does not enable full Unicode in identifiers by default.
```

Comment:
In presence of `-gnatW8` no other switches is required to have non-ASCII identifiers. So `-gnatiw` only required if there is no `-gnatW8`.

Reply:
Right, this was a leftover.

Reply:
Actually that was already removed in a later commit. Better if you check the changes as a whole instead of commit-by-commit.